### PR TITLE
generate mailers

### PR DESCRIPTION
**Pull request by drujensen**
Here is the procedure to test mailers:

1. create a new project
```
amber new blog --deps
```
2. generate scaffolding for a user
```
amber g scaffold user name:string email:string
```
3. generate a welcome mailer
```
amber g mailer welcome name:string email:string
```
4. send the email when a new user is created by adding the `deliver` to the create method
```
  def create
  user = User.new(params.to_h.select(["name", "email"]))

    if user.valid? && user.save
      WelcomeMailer.new.deliver(name: user.name.not_nil!, email: user.email.not_nil!)
      flash["success"] = "Created User successfully."
      redirect_to "/users"
    else
      flash["danger"] = "Could not create User!"
      render("new.slang")
    end
  end
```
5. launch application using docker-compose
```
dc up -d
```
6. create a new user at `open http://localhost:3000/users/new`

7. check mail catcher that an email was created `open http://localhost:1080`

### CODE REVIEW

**Comment by elorest** [comment]
Appropriate that the PR # is 25 lol.

**Comment by drujensen** [comment]
That's kinda freaky.

File: src/templates/mailer/src/mailers/{{name}}_mailer.cr.ecr [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+class <%= @name.capitalize %>Mailer < Quartz::Mailer
+  def initialize
+    super
+    from "from@example.com"
+  end
+
+  def deliver(<%= @fields.map{|f| "#{f.name} : #{f.cr_type}"}.join(", ") %>)
+    to "to@example.com"
+    subject "Example Subject"
+    body render("mailers/<%= @name %>_mailer.<%= @language %>", "mailer.<%= @language %>")
```
Comment (elorest) [inline src/templates/mailer/src/mailers/{{name}}_mailer.cr.ecr]: Glad you got it to work with my render method. 👍

File: src/amber_cmd/version.cr [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module Amber::CMD
-  VERSION = "0.1.13"
+  VERSION = "0.1.14"
```
Comment (eliasjpr) [inline src/amber_cmd/version.cr]: Have we bump the version? Or should we be bumping versions with each PR?

File: src/templates/mailer/src/mailers/{{name}}_mailer.cr.ecr [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require "quartz-mailer"
+
+class <%= @name.capitalize %>Mailer < Quartz::Mailer
```
Comment (eliasjpr) [inline src/templates/mailer/src/mailers/{{name}}_mailer.cr.ecr]: Should we be generating a view file for the Mailer?

**Comment by eliasjpr** [comment]
An issues was opened for this https://github.com/Amber-Crystal/amber_cmd/issues/27

File: src/amber_cmd/version.cr (hunk 2 above)
Comment (elorest) [inline src/amber_cmd/version.cr]: I think so. :+1:

File: src/templates/mailer/src/mailers/{{name}}_mailer.cr.ecr [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  def initialize
+    super
+    from "from@example.com"
+  end
+
+  def deliver(<%= @fields.map{|f| "#{f.name} : #{f.cr_type}"}.join(", ") %>)
+    to "to@example.com"
+    subject "Example Subject"
+    body render("mailers/<%= @name %>_mailer.<%= @language %>", "mailer.<%= @language %>")
+    super()
```
Comment (eliasjpr) [inline src/templates/mailer/src/mailers/{{name}}_mailer.cr.ecr]: Is there a reason for the () here, I think it should be super as above to stay consistent unless there is a reason why is this way.

File: src/templates/template.cr [hunk 5]
```diff
@@ -53,6 +54,9 @@ module Amber::CMD
       when "job"
         puts "Rendering Job #{name}"
```
Comment (eliasjpr) [inline src/templates/template.cr]: some colorize here would be nice.

File: src/templates/template.cr [hunk 6]
```diff
@@ -53,6 +54,9 @@ module Amber::CMD
...
         Job.new(name, fields).render(directory, list: true, color: true)
+      when "mailer"
+        puts "Rendering Mailer #{name}"
```
Comment (eliasjpr) [inline src/templates/template.cr]: some colorize for the `#{name}` would be nice.

File: src/amber_cmd/version.cr (hunk 2 above)
Comment (eliasjpr) [inline src/amber_cmd/version.cr]: Sorry Im a little confused. I thought when we do the release is when we bump the version. Currently we are on 0.1.13

File: src/templates/mailer/src/mailers/{{name}}_mailer.cr.ecr (hunk 4 above)
Comment (drujensen) [inline src/templates/mailer/src/mailers/{{name}}_mailer.cr.ecr]: added to initialize to be consistent.  The () are required when inherited class has different signature.

File: src/templates/mailer/src/mailers/{{name}}_mailer.cr.ecr (hunk 1 above)
Comment (drujensen) [inline src/templates/mailer/src/mailers/{{name}}_mailer.cr.ecr]: @elorest I haven't solved this yet.  I will look into it further.

**Pull request merged**